core/modules/installer/installer.browser.pages.inc

```diff
@@ -0,0 +1,884 @@
+
+    foreach ($release_data['releases'] as $version => $release) {
+      $release_title = format_string("@project @version - @date", array(
+        '@project' => $project['title'],
+        '@version' => $release['version'],
+        '@date' => format_date($release['date'], 'custom', 'M j, Y'),
+        ));
+      if (isset($release['terms']['Release type']) && !empty($release['terms']['Release type'])) {
+        $release_title .= check_plain(" (" . implode(', ', $release['terms']['Release type']) . ")");
+      }
+      if (isset($release['release_link'])) {
+        $releases_list[$version] = l($release_title, $release['release_link'], array('html' => TRUE));
+      }
+      else {
+        $releases_list[$version] = $release_title;
+      }
+    }
+
+    $form['releases'][$project['name']]['project'] = array(
+      '#type' => 'value',
+      '#value' => $project,
+    );
+
+    $form['releases'][$project['name']]['release_name'] = array(
+      '#type' => 'radios',
+      '#title' => t('Select release for @project', array('@project' => $project['title'])),
+      '#options' => $releases_list,
+      '#default_value' => key($releases_list),
+      '#prefix' => '<div class="installer-browser-releases-wrapper installer-browser-release-' . $project['name'] . '">',
+      '#suffix' => '</div>',
+      '#attributes' => array(
+        'class' => array('installer-browser-releases-radios'),
+      ),
+      '#required' => TRUE,
+    );
+    $form['releases'][$project['name']]['selected_text'] = array(
+      '#type' => 'item',
+      '#prefix' => '<div class="installer-browser-selected-release installer-browser-selected-release-' . $project['name'] . '">',
+      '#suffix' => '</div>',
+      '#markup' => reset($releases_list),
+    );
+    if (isset($project_data['recommended'])) {
+      // If there is a recommended release set, then only show it and show the
+      // jQuery link.
+      $recommended_releases = array();
+      $recommended_releases[$project_data['recommended']] = $releases_list[$project_data['recommended']];
+      $form['releases'][$project['name']]['release_name']['#default_value'] = $project_data['recommended'];
+      $form['releases'][$project['name']]['selected_text']['#markup'] = $releases_list[$project_data['recommended']];
+    }
+    if (count($releases_list) > 1) {
+      $form['releases'][$project['name']]['selected_text']['#markup'] .=
+        " (<span class='installer-browser-show-releases-link' rel='" . $project['name'] . "'>" . t('change release') . "</span>)";
+    }
+  }
+
+  // If there is nothing to install, go to the enable page.
+  if (empty($form['releases'])) {
+    backdrop_set_message(t('No releases data found for any of the selected projects.'), 'warning');
+    backdrop_goto('admin/modules/installer-browser/install/enable');
+  }
+
+  $form['backup_warning'] = array(
+    '#type' => 'markup',
+    '#markup' => t('Back up your database and site before you continue. !link.',
```

Comment:
Instead of doing this as markup on the page, let's do a `backdrop_set_message` of type `warning`.  It's too easy to overlook as-is.